rclcpp_lifecycle/test/test_lifecycle_node.cpp

```diff
@@ -142,6 +150,22 @@ TEST_F(TestDefaultStateMachine, empty_initializer) {
   EXPECT_EQ(State::PRIMARY_STATE_UNCONFIGURED, test_node->get_current_state().id());
 }
 
+TEST_F(TestDefaultStateMachine, empty_initializer_rcl_errors) {
+  {
+    auto patch = mocking_utils::patch_and_return(
+      "lib:rclcpp_lifecycle", rcl_lifecycle_state_machine_init, RCL_RET_ERROR);
+    EXPECT_THROW(
+      std::make_shared<EmptyLifecycleNode>("testnode"),
+      std::runtime_error);
```

Comment:
How are you checking for the returned throw? probably this is some sort of `rclcpp::exception::badArgument` exception

Reply:
It's not an invalid argument. If `rcl_lifecycle_state_machine_init` returns a non-OK code, it will just throw a `std::runtime_error`. https://github.com/ros2/rclcpp/blob/0313417f0213f1741465978c91ff18ba4f99ab55/rclcpp_lifecycle/src/lifecycle_node_interface_impl.hpp#L90-L105